src/components/ebay-progress-bar-expressive/test/test.browser.js

```diff
@@ -0,0 +1,63 @@
+import { expect, use } from "chai";
+import chaiDom from "chai-dom";
+import { composeStories } from "@storybook/marko";
+import { render, cleanup, waitFor } from "@marko/testing-library";
+import * as stories from "../progress-bar-expressive.stories";
+const { WithMessages, MediumSize } = composeStories(stories);
+
+use(chaiDom);
+afterEach(cleanup);
+
+/** @type import("@marko/testing-library").RenderResult */
+let component;
+
+describe("progress-bar-expressive", () => {
+    describe("given an expressive progress bar with messages", () => {
+        beforeEach(async () => {
+            component = await render(WithMessages);
+        });
+
+        it("then cycles through all messages", async () => {
+            const statusEl = component.getByRole("status");
+            const messages = WithMessages.args.messages;
+
+            // go through all messages once and loop back to first one
+            for (let i = 0; i <= messages.length; i++) {
+                const message = messages[i % messages.length];
+                await waitFor(
```

Comment:
This is a long-running test (several seconds). Could use [SinonJS fake timers](https://sinonjs.org/releases/latest/fake-timers/) to simulate and speed up